satpy/tests/reader_tests/test_msi_ec_l1c.py

```diff
@@ -0,0 +1,164 @@
+# Create 384 values for SOL_IRRAD
+SOL_IRRAD = SOL_IRRAD.reshape((-1, 1)) * np.ones(N_COLS, dtype=np.float32).reshape((1, -1))
+DIMLIST = np.ones((N_BANDS, N_SCANS, N_COLS))
+
+
+class FakeHDF5FileHandler2(FakeHDF5FileHandler):
+    """Swap-in HDF5 File Handler."""
+
+
+    def _setup_test_data(self, N_BANDS, N_SCANS, N_COLS):
+        # Set some default attributes
+        data = {
+            "ScienceData/pixel_values":
+                xr.DataArray(
+                    da.ones((N_BANDS, N_SCANS, N_COLS), chunks=1024, dtype=np.float32),
+                    attrs={"units": "Wm-2 sr-1 or K", "DIMENSION_LIST": DIMLIST},
+                    dims=("band", "dim_2", "dim_1")),
+            "ScienceData/land_flag":
+                xr.DataArray(
+                    da.ones((N_SCANS, N_COLS), chunks=1024, dtype=np.uint16),
+                    attrs={"units": ""},
+                    dims=("along_track", "across_track")),
+            "ScienceData/solar_azimuth_angle":
+                xr.DataArray(
+                    da.ones((N_SCANS, N_COLS), chunks=1024, dtype=np.float32),
+                    attrs={"units": "degrees"},
+                    dims=("along_track", "across_track")),
+            "ScienceData/longitude":
+                xr.DataArray(
+                    da.ones((N_SCANS, N_COLS), chunks=1024, dtype=np.float32),
+                    attrs={"units": "degrees"},
+                    dims=("along_track", "across_track")),
+            "ScienceData/latitude":
+                xr.DataArray(
+                    da.ones((N_SCANS, N_COLS), chunks=1024, dtype=np.float32),
+                    attrs={"units": "degrees"},
+                    dims=("along_track", "across_track")),
+            "ScienceData/solar_spectral_irradiance":
+                xr.DataArray(
+                    da.array(SOL_IRRAD),
+                    attrs={"units": "W m-2"},
+                    dims=("band", "across_track")),
+        }
+
+        return data
+
+    def get_test_content(self, filename, filename_info, filetype_info):
+        """Mimic reader input file content."""
+        test_content = self._setup_test_data(N_BANDS, N_SCANS, N_COLS)
+        return test_content
+
+
+class ECMSIL1CTester:
+    """Test MSI/EarthCARE L1C Reader."""
+
+    def setup_method(self):
+        """Wrap HDF5 file handler with our own fake handler."""
+        from satpy._config import config_search_paths
+        from satpy.readers.msi_ec_l1c_h5 import MSIECL1CFileHandler
+        self.reader_configs = config_search_paths(os.path.join("readers", self.yaml_file))
+        # http://stackoverflow.com/questions/12219967/how-to-mock-a-base-class-with-python-mock-library
+        self.p = mock.patch.object(MSIECL1CFileHandler, "__bases__", (FakeHDF5FileHandler2,))
+        self.fake_handler = self.p.start()
+        self.p.is_local = True
+
+    def teardown_method(self):
+        """Stop wrapping the HDF5 file handler."""
+        self.p.stop()
```

Comment:
Nowadays we try to create stub files using fixtures instead of mocking, to you think this is something we could do here?

Reply:
Could you point me to a test doing that? It is often confusing to know which way is "the good way" :-)

Reply:
Here an example, I hope it's good enough https://github.com/pytroll/satpy/blob/cd201899da539642e8a3131333583fcfe92557ad/satpy/tests/reader_tests/test_insat3d_img_l1b_h5.py#L91-L102
:D